ndk/src/hardware_buffer_format.rs

```diff
@@ -2,121 +2,63 @@
 //!
 //! [`AHardwareBuffer_Format`]: https://developer.android.com/ndk/reference/group/a-hardware-buffer#ahardwarebuffer_format
 
+use num_enum::{FromPrimitive, IntoPrimitive};
+
 /// Buffer pixel formats.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, FromPrimitive, IntoPrimitive)]
+#[non_exhaustive]
 #[allow(non_camel_case_types)]
 pub enum HardwareBufferFormat {
```

Comment:
added `non_exhaustive` here, but still on the fence whether that makes sense given that the `Unknown` type is **not** opaque. There will be ambiguity when we add new variants in non-breaking releases (assuming we can even generate them non-breaking in `ndk-sys` in the first place).

Reply:
hmmmm, it's a bit thorny yeah.

It conceptually makes sense to be non_exhaustive to allow extension without a breaking release but the `Unknown()` catch all kind of serves a similar purpose to the `non_exhaustive`.

Imagining having an exhaustive match of all know formats something like:
```rust
match format {
     Foo => { /* stuff */ },
     // <snip> exhaustive list of all (currently) known format

     Unknown(fmt) => { log::warn!("unknown format {fmt}") },
     fmt => { panic!("really unknown format {fmt}") }
}
```

the `#[non_exhaustive]` leads you to effectively have two catch all cases, and in this case a "non-breaking" release that would add a new format would suddenly cause the code above to panic.

It feels like maybe non_exhaustive isn't necessary but it could be considered semver compatible to extend the enum anyway because it doesn't break the API when the enum inherently has exhaustive coverage of all possible values.

Reply:
Nit: your `really unknown format` in reality is `ndk crate mapped a new format that we don't match on yet`.

Reply:
But anyway, having the two cases is effectively why one might want to make `Unknown` as opaque so that it becomes impossible to match on... Or to have `Unknown(u32)` _without `non_exhaustive`_ so that additions (that might have previously ended up in `Unknown(u32)` next to actual private/vendor values) will fail to compile rather than causing ambiguity.

Reply:
Removed it again, so that we don't have this duplicate ambiguity. Going forward we'll **either** (== mutually exclusive) have:

1. `#[num_enum(catch_all)] Unknown(variant)` publicly available;
2. `#[non_exhautive]`, and no publicly available catch-all (or a non-opaque one);
3. Neither of the two, mainly for "simpler" `enum`s that are unlikely to be extended and only used as function argument (i.e. user just needs to pass us a value, no need to `match` on it).